examples/test_examples.py

```diff
@@ -556,9 +563,178 @@ def test_text_to_image_checkpointing_use_ema(self):
             pipe = DiffusionPipeline.from_pretrained(tmpdir, safety_checker=None)
             pipe(prompt, num_inference_steps=2)
 
-            # check old checkpoints do not exist
-            self.assertFalse(os.path.isdir(os.path.join(tmpdir, "checkpoint-2")))
+            self.assertEqual(
+                set(os.listdir(os.path.join(tmpdir, "checkpoints"))),
+                {
+                    # no checkpoint_1 -> check old checkpoints do not exist
+                    "checkpoint_1",
+                    # check new checkpoints exist
+                    "checkpoint_2",
+                },
+            )
+
+    def test_text_to_image_checkpointing_resume_from_non_end_checkpoint(self):
```

Comment:
Since we modify the logic that looks for a checkpoint directory to resume from, we check that newly created checkpoints continuing from that location are created as expected